opm/autodiff/BlackoilDetails.hpp

```diff
@@ -156,6 +156,51 @@ namespace detail {
             }
         }
 
+        /// \brief Get the number of local interior cells in a grid.
+        /// \tparam The type of the DUNE grid.
+        /// \param grid The grid which cells we count
+        /// \return The number of interior cell in the partition of the
+        /// grid stored on this process.
+        template<class Grid>
+        std::size_t countLocalInteriorCells(const Grid& grid)
+        {
+            if ( grid.comm().size() == 1)
+            {
+                return grid.size(0);
+            }
+            std::size_t count = 0;
+            const auto& gridView = grid.leafGridView();
+            for(auto cell = gridView.template begin<0>(),
+                    endCell = gridView.template end<0>();
+                cell != endCell; ++cell)
```

Comment:
Why not auto cell = gridView.template begin<0, Interior_Partition> ? Then the if could be avoided. But the code is fine like it is too.